luaui/Widgets/cmd_nano_range_check.lua

```diff
@@ -0,0 +1,277 @@
+    local is_changed = false
+    local is_first_cmd = true
+    local cmds = Spring.GetUnitCommands(uID, -1)
+    local new_cmds = {}
+    for i = #cmds, 1, -1  do
+        local cmd = cmds[i]
+        if isValidCommandID(cmd.id) then
+            local tuID = cmd.params[1]
+            local tx, ty, tz = Spring.GetUnitPosition(tuID)
+            if tx == nil then break end
+            local distance = calculateDistance2D(x, z, tx, tz)
+            local radius = getUnitRadius(tuID)
+            if distance < build_distance + radius then -- BP uses build_distance + radius (sphereical shape of model)
+                cmd.options.shift = not is_first_cmd
+                is_first_cmd = false
+                table.insert(new_cmds, {cmd.id, cmd.params, cmd.options}) -- building cmdArr
+            else
+                is_changed = true
+            end
+        end
+    end
+    if is_changed then
+        if #new_cmds > 0 then
+            table.insert(new_orders, {uID, new_cmds}) -- schedule for new order
+        else
+            table.insert(to_be_cleared, uID) -- schedule for clear
+        end
+    end
+end
+
+-- Applies all stop orders to units in the `to_be_cleared` stack.
+local function applyStopOrders()
+    if #to_be_cleared <= 0 then return end
+    Spring.GiveOrderToUnitArray(to_be_cleared, CMD.STOP, {}, {})
+    command_budget = command_budget - #to_be_cleared
+    to_be_cleared = {}
+end
+
+-- Processes new orders that have been scheduled for execution in the `new_orders` stack.
+local function processNewOrders()
+    if #new_orders == 0 and #to_be_cleared > 0 then return end
+    for i = #new_orders, 1, -1 do
+        if command_budget <= 0 then break end
+        Spring.GiveOrderArrayToUnit(new_orders[i][1], new_orders[i][2])
+        table.remove(new_orders, i) -- pop
+        command_budget = command_budget - 1
+    end
+end
+
+-- Processes the turret range check after a certain number of frames (controlled by `DELAY`).
+-- It iterates over the current list of nano turrets and checks their range to ensure they are not assigned
+-- to constructions out of reach.
+-- If any changes are detected, it schedules new orders or/and clears existing ones.
+local function processTurretRange()
+    if counter >= DELAY then
+        counter = 0
+        for _, tower in ipairs(current_towers) do
+            checkTurretRange(tower)
+        end
+        flag_processed = true
+    end
+end
+
+-- Handles the flow control. If no command is scheduled, it disables the listener for the gameframe loop,
+-- else it keeps calling the `processTurretRange` function, in case anything changed during re-ordering.
+local function handleFlowControl()
+    if #to_be_cleared == 0 and #new_orders == 0 then
+        if flag_processed then
+            flag_listening = false
+        else
+            processTurretRange()
+        end
+    end
+end
```

Comment:
Why does the listener need to be disabled if there is nothing to do? The performance impact of iterating over an empty table is negligible.